### PR TITLE
Add some problem to bst

### PR DESCRIPTION
**Pull request by danghai**
PR to add some problem to bst along with test case. A commit fix typo in `bst.py`

### CODE REVIEW

File: tree/bst/height.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+"""
+import unittest
+from bst import Node
+from bst import bst
+
+def height(root):
+    if root is None:
+        return 0
+    else:
+        return max(1 + height(root.left), 1 + height(root.right))
```
Comment (tratneshsingh) [inline tree/bst/height.py]: Please change this to the following:
return 1 + max(height(root.left), height(root.right))

File: tree/bst/num_empty.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+"""
+import unittest
+from bst import Node
+from bst import bst
+
+def num_empty(root):
+    if root is None:
+        return 1
+    elif root.left is None and root.right is None:
```
Comment (tratneshsingh) [inline tree/bst/num_empty.py]: Lines from 28-33 are redundant as the condition in 26 returns 1 if root is None.

Comment (danghai) [reply tree/bst/num_empty.py]: Yeah, thanks

File: tree/bst/height.py (hunk 1 above)
Comment (tratneshsingh) [reply tree/bst/height.py]: Try doing
``` python

# If you mean continuation, use the following.
return A + B + \
  C + D

# The below returns 'abc'
return reduce(operator.concat, ['a', 'b', 'c'])

# There are other ways to do the continuation : 'abc'
return ''.join(['a', 'b', 'c'])
```

Comment (goswami-rahul) [reply tree/bst/height.py]: @danghai Or you can simply do
```python
return (A + B
            + C + D)
```

File: tree/bst/num_empty.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+"""
+import unittest
+from bst import Node
+from bst import bst
+
+def num_empty(root):
+    if root is None:
+        return 1
+    elif root.left is None and root.right:
```
Comment (tratneshsingh) [inline tree/bst/num_empty.py]: As line 28 to line 31 are redundant, please remove

**Pull request merged**